scripts/cmake/depends/gnuefi.cmake

```diff
@@ -0,0 +1,43 @@
+#
+# Bareflank Hypervisor
+# Copyright (C) 2015 Assured Information Security, Inc.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+if(ENABLE_BUILD_EFI AND NOT WIN32)
+    message(STATUS "Including dependency: gnuefi")
+
+    download_dependency(
+        gnuefi
+        URL         ${GNUEFI_URL}
+        URL_MD5     ${GNUEFI_URL_MD5}
+    )
+
+    add_dependency(
+        gnuefi userspace
```

Comment:
How does this work? GNU efi is built for the userspace prefix, but depends on stuff in the VMM prefix?

Reply:
This is a very good question and should have a little RFC.  gnu-efi uses some C library headers which it normally gets from the system you're building things on.  For build portability and consistency, I have it use the ones we use in our project.  They're only installed to the vmm prefix though, and because gnu-efi isn't part of the vmm, using headers from the vmm prefix seems like bad architecture.  Merely copying some clib headers from vmm to userspace prefix doesn't isn't any better though.  The project already relies on the system having various tools and libraries present, so maybe this attempt at portability should be scrapped.  Need some input